### DISCUSSION TITLE
Custom deformable mesh object interaction

### DISCUSSION THREAD

**Discussion by jpmartin42** [isaac-sim/IsaacLab#1447] [Q&A]
I'm looking into ways to integrate custom deformable terrain morphologies into Isaac Lab for learning in deformable terrain environments. I've been reading through simulators with fast support for multiple meshes being run simultaneously, and it seems that Isaac Lab has [some capacity for custom meshes using PhysX](https://isaac-sim.github.io/IsaacLab/main/source/tutorials/01_assets/run_deformable_object.html#the-code). However, the mesh here is treated as a physics object using default rules, whereas I'm looking into ways to add my own custom mesh that behaves as deformable terrain would.

Are there any resources/examples for making custom rules for object-mesh and node-node interaction within a mesh? I want to apply my own laws to make the mesh behave with a certain compliance and permanent deformation after being interacted with.

**Comment by StrainFlow** [comment]
I would take a look at Warp for something like this: 

https://docs.omniverse.nvidia.com/extensions/latest/ext_warp.html
https://github.com/NVIDIA/warp
